src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -1010,8 +1012,19 @@ class RandomStringFromDataset : public Generator<std::string> {
 
     std::string evaluate() {
         auto dataset = _datasets.getDatasetForPath(_path);
-        auto distribution = boost::random::uniform_int_distribution<size_t>{0, dataset.size() - 1};
-        return dataset[distribution(_rng)];
+        if(_sequential) {
```

Comment:
This probably doesn't matter, but could we move this conditional out of the evaluate function? It's either going to always be true or always false, and evaluate is on the critical path. 

This isn't likely worth much effort, but if we have any examples or easy way of evaluating this early and/or using two different classes, that would be pretty slick. And if it is a lot of effort, please just say so and let's move on with our days.

Reply:
Done; split into two classes.